utils/build-script-impl

```diff
@@ -1826,6 +1845,28 @@ for deployment_target in "${NATIVE_TOOLS_DEPLOYMENT_TARGETS[@]}" "${CROSS_TOOLS_
                 # Foundation builds itself and doesn't use cmake
                 continue
                 ;;
+			libdispatch)
+                LIBDISPATCH_BUILD_DIR=$(build_directory ${deployment_target} ${product})
+
+				set -x
+				if [[ ! -f "${LIBDISPATCH_BUILD_DIR}"/config.status ]]; then
+					# First time building; need to run autotools and configure
+					mkdir -p "${LIBDISPATCH_BUILD_DIR}"
+					pushd "${LIBDISPATCH_SOURCE_DIR}"
+					autoreconf -fvi
+					popd
+					pushd "${LIBDISPATCH_BUILD_DIR}"
+					"${LIBDISPATCH_SOURCE_DIR}"/configure --prefix="${INSTALL_DESTDIR}"/"${INSTALL_PREFIX}"
```

Comment:
I put in some of the plumbing at higher levels of the scripts to support multiple build types following what is done for foundation, but right now it isn't being used by the autotools driven build.  I think it makes sense to eventually have multiple build types, but I'm not 100% sure exactly how it will be passed down, so leaving for a latter commit.